### PR TITLE
Use build_runner

### PR DESCRIPTION
**Pull request by daavoo**

### CODE REVIEW

File: mmcls/core/evaluation/eval_hooks.py [hunk 1]
```diff
@@ -12,21 +12,30 @@ class EvalHook(Hook):
         interval (int): Evaluation interval (by epochs). Default: 1.
     """
 
-    def __init__(self, dataloader, interval=1, **eval_kwargs):
+    def __init__(self, dataloader, interval=1, by_epoch=False, **eval_kwargs):
```
Comment (hellock) [inline mmcls/core/evaluation/eval_hooks.py:15]: Better to set the default value to True to keep consistent with the old behavior?

File: docs/tutorials/finetune.md [hunk 2]
```diff
@@ -75,8 +75,8 @@ optimizer_config = dict(grad_clip=None)
 lr_config = dict(
     policy='step',
     step=[15])
-total_epochs = 20
-log_config = dict(interval=100)
+ runner = dict(type='EpochBasedRunner', max_epochs=200)
+ log_config = dict(interval=100)
```
Comment (hellock) [inline docs/tutorials/finetune.md:79]: Useless space

File: mmcls/apis/train.py [hunk 3]
```diff
@@ -70,12 +71,26 @@ def train_model(model,
...
-        meta=meta)
+
+    if cfg.get('runner') is None:
+        cfg.runner = {
+            'type': 'EpochBasedRunner',
+            'max_epochs': cfg.total_epochs
+        }
+        warnings.warn(
+            'config is now expected to have a `runner` section, '
+            'please set `runner` in your config.', DeprecationWarning)
```
Comment (hellock) [inline mmcls/apis/train.py:82]: `DeprecationWarning` is filtered by default and will not be raised, we may just use UserWarning.

**Comment by ycxioooong** [comment]
Hi @daavoo,  since we have merged a new PR(#61) that adds linting into the workflow, so it infects your current code. Could you please pull the latest code to your forked repo and recommit? I think it will fix lint failure.

**Comment by daavoo** [comment]
> Hi @daavoo, since we have merged a new PR(#61) that adds linting into the workflow, so it infects your current code. Could you please pull the latest code to your forked repo and recommit? I think it will fix lint failure.

Fixed. Thanks!

**Review by ycxioooong — APPROVED** [review]

**Pull request merged**